### PR TITLE
Integrations overview

### PR DESCRIPTION
**Pull request by ES-Alexander**
### This PR:
- [x] add mermaid integration
   - [x] support light/dark transitions
- [x] create overview flowchart

### Future PR(s):
- add description of overview flowchart
- create vehicle components diagram
- enable user to change zoom level (with slider / +/- buttons)
- Action to automatically update mermaid file from CDN?

### CODE REVIEW

File: templates/shortcodes/mermaid.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+</div>
+
+<!-- Use latest mermaid renderer from the online CDN by default, -->
+<script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>
+<!-- but fall back to a local version if the CDN load failed (e.g. if using docs offline) -->
+<script window.mermaid || document.write('<script src="/js/mermaid.min.js">\x3C/script>')></script>
+
+<!-- when this javascript is loaded and initialized it will look for divs with the class mermaid.
+     it will replace the content of those divs with the generated svg -->
+<script>mermaid.initialize({startOnLoad:true});</script>
```
Comment (patrickelectric) [inline templates/shortcodes/mermaid.html:22]: Doesn't it make more sense to put it under `header.html` or `base.html` ?
With that the script will be included only once when the shortcode is used and not every time.

Comment (ES-Alexander) [reply templates/shortcodes/mermaid.html:22]: I considered that, but then it's also included on every page without a diagram, right? I'm assuming that most pages won't have diagrams, and that pages that do have diagrams will generally only have one (or at least a small number).

**Comment by ES-Alexander** [comment]
> The size appears a bit weird here

Not sure what the best approach is for the size - it's constrained to the width between where normal pages have a sidebar (on the left) and a table of contents (on the right). At the moment it's possible to zoom the page to see more clearly, but it would be nice if there's a way to zoom just the diagram.

> dark/light theme are working

I was hoping to have the diagram generated with mermaid's dark theme when the page is in dark mode. At the moment the docs has light/dark mode which works fine, but the diagram is always just mermaid's default "neutral" theme, which is a bit unpleasant with the dark mode of the docs (even with the lines changed to work with both themes). The lines are the biggest issue though, and I could also change the other default colours to better work with both docs themes if it's not simple to use mermaid's theme options, I'd just prefer not to need to, especially since mermaid already has themes.

Thoughts?

**Comment by ES-Alexander** [comment]
latest state: 

https://user-images.githubusercontent.com/25898329/201448810-6fbb6701-51ba-43bf-a182-b55647ed726a.mov

**Comment by ES-Alexander** [comment]
Realised I hadn't rebased over the last couple of submodule updates in `main`, so I did that.

File: templates/shortcodes/mermaid.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+<div style="overflow:auto;">
+<pre class="mermaid pre" style="width:{%if width %}{{width}}{% else %}100%{% endif %};">
+{{ data }}
+</pre>
+</div>
+
+<!-- when this javascript is loaded and initialized it will look for divs with the class mermaid.
+     it will replace the content of those divs with the generated svg -->
+<script type="module">
+ // sourced from https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs
```
Comment (patrickelectric) [inline templates/shortcodes/mermaid.html:20]: This should be in the commit body since the code can be replaced and new versions could be used.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**